### PR TITLE
Refactor logging component

### PR DESCRIPTION
**Pull request by cthulhu-rider**
* relates #696

The most meaningful changes are placed in
- `pkg/util/logger` package
- `cmd/neofs-node/config.go`

All recent changes are just routine.

### CODE REVIEW

**Review by acid-ant — APPROVED** [review]

File: cmd/neofs-node/config.go [hunk 1]
```diff
@@ -873,51 +848,51 @@ func (c *cfg) configWatcher(ctx context.Context) {
...
-			// Storage Engine
+	err := c.cfgObject.cfgLocalStorage.localStorage.Reload(rcfg)
+	c.exitOnFailedConfigApply("storage engine", err, exitOnErr)
 
-			var rcfg engine.ReConfiguration
-			for _, optsWithID := range c.shardOpts() {
-				rcfg.AddShard(optsWithID.configID, optsWithID.shOpts)
-			}
+	// Logger
+	c.dynamicConfiguration.logger.SetLevel(c.applicationConfiguration.LoggerCfg.level)
```
Comment (carpawell) [inline cmd/neofs-node/config.go:878]: why without `Reload`?

File: pkg/util/logger/config.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Config provides dynamic logging configuration. See methods for details.
+//
+// For correct operation, Config instance MUST be constructed initialized
+// using Logger.Init.
+type Config struct {
```
Comment (carpawell) [inline pkg/util/logger/config.go:14]: so have you rejected the idea with `Reload` method from #1815?

**Review by carpawell — COMMENTED** [review]
Suggest merging that ASAP, too many conflicts.

File: cmd/neofs-node/config.go [hunk 3]
```diff
@@ -88,7 +86,7 @@ type applicationConfiguration struct {
 	_read bool
 
 	LoggerCfg struct {
-		level string
+		level logger.Level
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply cmd/neofs-node/config.go:89]: Why? Config die already tells the raw structure.

File: cmd/neofs-node/config.go [hunk 4]
```diff
@@ -501,6 +511,9 @@ var persistateSideChainLastBlockKey = []byte("side_chain_last_processed_block")
 
 func initCfg(appCfg *config.Config) *cfg {
 	c := &cfg{}
+	c.internals.init(appCfg)
+
+	c.internals.log.Init(&c.dynamicConfiguration.logger)
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply cmd/neofs-node/config.go:516]: This seems strange because logger isn't really an internal entity, or I don't understand what internals means.

**Comment by cthulhu-rider** [comment]
Merging is blocked by https://github.com/nspcc-dev/neofs-node/milestone/49.

**Comment by cthulhu-rider** [comment]
Stale, leave the branch.

**Pull request closed without merging**